tests/test_dse.py

```diff
@@ -391,3 +391,23 @@ def test_makeit_ssa(exprs, exp_u, exp_v):
 
     assert np.all(u.data == exp_u)
     assert np.all(v.data == exp_v)
+
+
+def test_time_dependent_split():
+    grid = Grid(shape=(4, 4))
+    u = TimeFunction(name='u', grid=grid, time_order=2, space_order=1, save=3)
+    v = TimeFunction(name='v', grid=grid, time_order=2, space_order=0, save=3)
+
+    # The second equation needs a full loop over x/y for u then
+    # a full one over x.y for v
+    eq = [Eq(u.forward, 2 + grid.time_dim),
+          Eq(v.forward, u.forward.dx + u.forward.dy + 1)]
+    op = Operator(eq, dse='advanced', dle='basic')
+
+    trees = retrieve_iteration_tree(op)
+    assert len(trees) == 2
```

Comment:
@fablup so mainly this test replace the tti one that checks that the loop is split for u and v (u needs to be updated everywhere first then v)
Unfortunately, it looks like the only reason the other test was passing is that the src term was between the two forcing it to be split. This one fails :( I'm looking into it.

Reply:
I tried it. I think this is just yet another example of "machines beating humans". 
Look at the iteration direction of x and y: they actually go backwards to turn the antidependence from the first to the second equation (IOW, the second equation needing information computed by the first one). So we shouldn't worry about this?

Reply:
if you use ``space_order=2`` in the definition of ``u``, then you **also** get a flow dependence, which will force the compiler to split the loops

Reply:
Wow indeed nice catch, that's quite amazing :D